src/types/Offer.ts

```diff
@@ -146,8 +147,15 @@ type Label = {
   excluded: boolean;
 };
 
-export const hasLegacyLocation = (offer) =>
-  offer?.location && !offer?.location?.['@id'];
+export const hasLegacyLocation = (offer: Offer) => {
+  if (!offer) return false;
+
+  if (parseOfferType(offer['@context']) === 'place') {
```

Comment:
I think there is also a isPlace utility function?

Reply:
If a isEvent function exists, you can check if (!isEvent(offer)) return false.

Then (offer as Event won't be needed)

Reply:
👍